### PR TITLE
Egoscale V3: Replace Zone with url endpoints

### PR DESCRIPTION
**Pull request by pierre-emmanuelJ**
# Description

Add zoned URL to replace standalone zones and useless `Environment`

### CODE REVIEW

File: v3/client.go [hunk 1]
```diff
@@ -9,30 +9,41 @@ import (
...
+	CHDk2  URL = "https://api-ch-dk-2.exoscale.com/v2"
+	DEFra1 URL = "https://api-de-fra-1.exoscale.com/v2"
+	DEMuc1 URL = "https://api-de-muc-1.exoscale.com/v2"
+	ATVie1 URL = "https://api-at-vie-1.exoscale.com/v2"
+	ATVie2 URL = "https://api-at-vie-2.exoscale.com/v2"
+	BGSof1 URL = "https://api-bg-sof-1.exoscale.com/v2"
 )
 
+// URLs represents a list of zoned URL.
+var URLs map[string]URL = map[string]URL{
```
Comment (aureliar8) [inline v3/client.go:32]:
```suggestion
var Zones map[string]URL = map[string]URL{
```
Maybe this is clearer for the variable name

**Review by aureliar8 — APPROVED** [review]

**Review by kobajagi — APPROVED** [review]

File: v3/client.go [hunk 2]
```diff
@@ -113,7 +117,6 @@ func NewClient(apiKey, apiSecret string, opts ...ClientOpt) (*Client, error) {
 		apiKey:          apiKey,
 		apiSecret:       apiSecret,
 		serverURL:       "https://api-ch-gva-2.exoscale.com/v2",
```
Comment (sauterp) [inline v3/client.go:119]:
```suggestion
		serverURL:       string(CHGva2),
```
didn't test this

**Review by sauterp — APPROVED** [review]

File: v3/client.go (hunk 2 above)
Comment (pierre-emmanuelJ) [reply v3/client.go:119]: I need to modify the generator, to change the default constructor logic, not the generated file. Let me try

Comment (pierre-emmanuelJ) [reply v3/client.go]: https://github.com/exoscale/egoscale/pull/610/commits/f2289d326c5869140b956050a4d2f74fe5ea35be

**Pull request merged**